### ISSUE TITLE
Question - about repo structure - enhancement perhaps?

### ISSUE THREAD

**Issue by cliffordfajardo** [Eugeny/tabby#2334]
Hey @Eugeny 
I've been using terminus for the past few weeks and I love! I've also been learning the codebase so I can start making my own addons

I noticed the terminus repo is structured like a mono repo, where you have packages split out nicely. For example:

```
app/
terminus-core
terminus-plugin-manager
terminus-serial
.....
```

Have you considered using yarn's native feature called "workspaces"?  It would eliminate quite of few of the build scripts and build specific code simply adding 1 additional properties to your package.json

```
"workspaces": [
  "terminus-core",
  "terminus-plugin-manager",
  "terminus-serial"
   ...
], 

```

Yarn workspaces handles all the symlinking between modules.
- [Yarn workspace VanillaJS Example ](https://www.youtube.com/watch?v=G8KXFWftCg0)
- [Yarn workspace vanilla typescript example](https://www.youtube.com/watch?v=EoqzOEZIzhg)
- [Yarn workspace with Lerna example](https://www.youtube.com/watch?v=p6qoJ4apCjA&t=200s)

```
terminus
├── terminus-terminal
 │   ├── index.js
 │   └── package.json
├── node_modules
 │   ├── terminus-terminal -> ../terminus-terminal
 │   └── termius-core -> ../terminus-core
├── terminus-core 
 │   ├── index.js
 │   └── package.json
├── package.json
└── yarn.lock
```

I recently found myself implementing a similar repo structure as yours for a work project, which led to look at yarn workspaces.

**Comment by cliffordfajardo** [comment]
Also, is there a discord chat server or anything similar for this project for collaboration/questions purposes?

Thanks!

**Comment by Eugeny** [comment]
I'm definitely open to improvements! There's some webpack glue magic in place to avoid listing common dependencies six times, and due to the plugin loading logic it's critical that Angular modules only exist once in `app/node_modules` and not `terminus-*/node_modules`. 
Another thing is there's a bunch of native module compilation involved in the build process and I'm not sure yet how Yarn 2 handles that given that the toolchain (node-gyp, electron-rebuild etc.) expect a node_modules folder.

In short, if you can pull this off, I'd love to merge!

**Comment by Eugeny** [comment]
As for communication, there'a Gitter chatroom (link in the README)

**Comment by geodic** [comment]
I don't think this issue should be marked as stale... This problem still makes packaging very difficult in NixOS: https://github.com/NixOS/nixpkgs/issues/233509. I have successfully been able to package this on NixOS with a lot of patchwork (https://github.com/NixOS/nixpkgs/pull/368048), but I don't think this is maintainable at all. For example, automatic package updates are currently impossible due to the many hash recomputations of each individual `yarn.lock`. Using the yarn `workspaces` feature will definitely make this easier. I personally don't use yarn and have no idea how the workspaces feature works... Are you still interested in this @cliffordfajardo?
